src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/ReferenceSearchValue.cs

```diff
@@ -21,11 +20,11 @@ public class ReferenceSearchValue : ISearchValue
         /// <param name="baseUri">The base URI of the resource.</param>
         /// <param name="resourceType">The resource type.</param>
         /// <param name="resourceId">The resource id.</param>
-        public ReferenceSearchValue(ReferenceKind referenceKind, Uri baseUri, ResourceType? resourceType, string resourceId)
+        public ReferenceSearchValue(ReferenceKind referenceKind, Uri baseUri, string resourceType, string resourceId)
         {
             if (baseUri != null)
             {
-                EnsureArg.IsNotNull(resourceType, nameof(resourceType));
+                EnsureArg.IsNotNullOrWhiteSpace(resourceType, nameof(resourceType));
```

Comment:
Should we also validate to make sure it's supported ResourceType?

Reply:
Added `ModelInfoProvider.EnsureValidResourceType`